app/controllers/api_controller.rb

```diff
@@ -10,11 +10,15 @@ def ensure_jwt
 
   def token_match?
     auth_header = request.headers['HTTP_AUTHORIZATION']
-    return false unless auth_header
     token = auth_header.split(' ').last
     secret = Rails.application.secrets.artsy_internal_secret
-    JWT.decode(token, secret)
-  rescue JWT::DecodeError
+    payload = JWT.decode(token, secret).first
+    validate_payload(payload)
+  rescue JWT::DecodeError, NoMethodError
```

Comment:
`NoMethodError` seems too generic here.

Reply:
I agree - any reason you didn't like the return false unless line @cavvia?

Reply:
I agree - Rubocop was giving me an [AbcSize](www.rubydoc.info/gems/rubocop/0.27.0/RuboCop/Cop/Metrics/AbcSize) error, though I may have fixed that with the `validate_payload` extraction too. So can re-introduce and see if that works.

Reply:
This is back now.